src/main/java/org/springframework/data/repository/core/support/RepositoryFactorySupport.java

```diff
@@ -534,6 +540,7 @@ protected void validate(RepositoryMetadata repositoryMetadata) {
 	public class QueryExecutorMethodInterceptor implements MethodInterceptor {
```

Comment:
Given that we're adding yet another class to this arrangement, does it make sense to extract `QEMI` into its own file and make `QMI` a nested class in that one?